src/main/java/com/spartronics4915/frc2020/subsystems/PanelRotator.java

```diff
@@ -1,16 +1,47 @@
 // I'm bad at naming things. Please come up with a better name...
 package com.spartronics4915.frc2020.subsystems;
 
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.ColorSensorV3;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import com.spartronics4915.frc2020.Constants;
 import com.spartronics4915.lib.subsystems.SpartronicsSubsystem;
 
+import 
+
+import edu.wpi.first.wpilibj.DigitalInput;
+
 public class PanelRotator extends SpartronicsSubsystem {
+    private final CANSparkMax mSpinMotor;
+    private final CANSparkMax mExtendMotor;
+    private final DigitalInput mBeamSensorUp;
+    private final DigitalInput mBeamSensorDown;
+    private final ColorSensorV3 mColorSensorRight;
+    private final ColorSensorV3 mColorSensorLeft;
+    private final I2C.Port i2cPort = 
 
     public PanelRotator() {
-        
+        mBeamSensorUp = new DigitalInput(Constants.kBeamSensorUpID);
+        mBeamSensorDown = new DigitalInput(Constants.kBeamSensorDownID);
+        mSpinMotor = new CANSparkMax(Constants.kSpinMotorID, MotorType.kBrushless);
+        mExtendMotor = new CANSparkMax(Constants.kExtendMotorID, MotorType.kBrushless);
+        mColorSensorLeft = new ColorSensorV3(Constants.kColorSensorLeftID);
+        mColorSensorRight = new ColorSensorV3(Constants.kColorSensorRightID);
     }
 
-    public void rotate(double number) {
+    public void raise() {
+        mExtendMotor.set(0.5);
+        if(!mBeamSensorUp.get())
+            mExtendMotor.set(0);
+    }
+    public void lower() {
+        mExtendMotor.set(-0.5);
+        if(mBeamSensorDown.get())
```

Comment:
Bleh, yeah... I think we're probably going to end up switching to an encoder. 

I know I said something different yesterday, but you should remove these checks and instead have a single boolean method for each sensor (ex. `getBeamSensorDown/Up`). Checking the sensor should be done in a command (as part of the `isFinished()` method).

Reply:
The documentation says the sensor returns `1` until broken, then returns `0`. It's up to you whether you want to keep that the same or flip it - either way, just make it clear.

Reply:
We are definitely switching to the internal encoder on the NEO 550. You will need to use a PD controller and do the appropriate conversions.